public/app/features/templating/template_srv.ts

```diff
@@ -124,6 +124,13 @@ export class TemplateSrv {
         }
         return value;
       }
+      case 'urlescape': {
+        // like glob, but url escaped
+        if (_.isArray(value)) {
+          return escape('{' + value.join(',') + '}');
+        }
+        return escape(value);
```

Comment:
think this should use encodeURIComponent

Reply:
encodeURIComponent does not encode "- _ . ! ~ * ' ( )", see https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

Reply:
well escape is not a function that is not recommended for use, only there for legacy reasons. what do you mean by urlescape? that the whole value is a complete url or just an url param? if just an url param you should use escapeURIComponent if it's a complete url you should use escapeURI

https://stackoverflow.com/questions/75980/when-are-you-supposed-to-use-escape-instead-of-encodeuri-encodeuricomponent

Reply:
I agree, but escapeURIComponent() leaves some characters plain, that must be encoded for icingaweb to play nicely.  So is your concern about the deprecated escape() call or about having a format function that escapes all special chars?

Reply:
> So is your concern about the deprecated escape() call or about having a format function that escapes all special chars? 

I would prefer to use the standard one and not the deperecated one

Reply:
Ok, I've reworked this to avoid the legacy escape() call. I had to create a dedicated function hence it shall percent encode all characters so they can be safely passed as a query arg value. I've also renamed the formatter to emphasize this difference.